posts/napari-programmatically-stitching/stitching-programmatically.qmd

```diff
@@ -0,0 +1,539 @@
+
+```{python}
+#| label: confirm napari
+#| eval: false
+
+import napari
+from skimage import data
+
+viewer, image_layer = napari.imshow(data.astronaut())
+```
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-confirmation"
+#| fig-cap: "Optionally launching a built-in image to ensure that napari was successfully installed. Shown here is an image of [Elleen Collins](https://www.nasa.gov/former-astronaut-eileen-collins/){target='_blank'}."
+
+knitr::include_graphics("./figures/fig-confirmation.png")
+```
+
+## Installing `napari-cosmx`
+
+Now that we have our virtual environment installed, activated, and
+napari installed, we need to install the plugin. The file (and license)
+can be found in [asests/napari-cosmx
+releases](https://github.com/Nanostring-Biostats/CosMx-Analysis-Scratch-Space/tree/Main/assets/napari-cosmx%20releases){target="_blank"}
+in the source code of our scratch space repository. Download this `whl`
+file, return to bash/cmd, and install it like this:
+
+::: {.panel-tabset group="language"}
+### Mac/Unix
+
+```{bash}
+#| label: install napari-cosmx mac
+#| eval: false
+
+pip install /path/to/your/napari_CosMx-0.4.17.0-py3-none-any.whl
+```
+
+### Windows
+
+```{bash}
+#| label: install napari-cosmx windows
+#| eval: false
+
+pip install "C:\path\to\your\napari_CosMx-0.4.17.0-py3-none-any.whl"
+
+```
+:::
+
+There are a few package scripts that are installed within your virtual
+environment. The two we'll use on today is `stitch-images` and
+`read-targets`.
+
+# Stitching a single slide in python {#sec-single-slide}
+
+Before we stitch multiple slides, let's stitch a single slide within
+python. In the code below, we point python to the location of our raw
+data slide folder. This folder will have subfolders `CellStatsDir`,
+`RunSummary`, and `AnalysisResults`.
+
+In the user-defined options section of this script, be sure to configure
+the paths as needed.
+
+```{python}
+#| label: stitch one slide
+#| eval: false
+
+import numpy
+import pandas as pd
+import os
+from os import listdir
+from glob import iglob
+
+### User-defined options
+
+## Path to your slide folder
+# SlideDir = "path/to/your/slide/raw/data" # Mac/Unix OR
+# SlideDir = r"C:\path\to\your\slide\raw\data" # Windows
+
+## Output path
+# SlideOutputDir = "/path/to/new/directory/to/store/napari/files" # Mac/Unix OR
+# SlideOutputDir = r"C:\path\to\new\directory\to\store\napari\files" # Windows
+
+### Processing
+
+if not os.path.exists(SlideOutputDir):
+  print("Creating output folder")
+  os.makedirs(SlideOutputDir)
+
+CellStatsDir = os.path.join(SlideDir, "CellStatsDir")
+RunSummaryDir = os.path.join(SlideDir, 'RunSummary')
+AnalysisDirParent = os.path.join(SlideDir, 'AnalysisResults')
+AnalysisDirSubBasename = [i for i in listdir(AnalysisDirParent) if not i.startswith('.')]
+AnalysisDir = os.path.join(AnalysisDirParent, AnalysisDirSubBasename[0])
+
+cmd_stitch = 'stitch-images -i "' + CellStatsDir + '" -f "' + RunSummaryDir + '" -o ' + SlideOutputDir
```

Comment:
This command did not work for me since the outputdir did not end up in quotes. The below worked:
cmd_stitch = 'stitch-images -i "' + CellStatsDir + '" -f "' + RunSummaryDir + '" -o "' + SlideOutputDir + '"'